frontend/src/Components/MatchTable/MatchTable.jsx

```diff
@@ -59,9 +59,12 @@ class MatchTable extends React.Component{
     };
 
     render() {
-
+        const sortedMatches = this.state.matches.sort(function(a, b) {
+            const order = a.user2_id === null ? -1 : 1;
```

Comment:
technically you can't join your own match, so one more filter before this should handle that